### PR TITLE
Kernel: Make sys$msync more posix compliant and reduce its IO heaviness

### PR DESCRIPTION
**Pull request by Hendiadyoin1**
This also adds some more FIXME's.
Note: These are sadly not directly tested, due to us not having any tests for msync.

#### Kernel: Don't rewrite the whole file on sys$msync

### CODE REVIEW

File: Kernel/Syscalls/mmap.cpp [hunk 1]
```diff
@@ -590,19 +590,36 @@ ErrorOr<FlatPtr> Process::sys$msyscall(Userspace<void*> address)
...
 }
 
-ErrorOr<FlatPtr> Process::sys$msync(Userspace<void*> address, size_t size, [[maybe_unused]] int flags)
+ErrorOr<FlatPtr> Process::sys$msync(Userspace<void*> address, size_t size, int flags)
 {
+    // Either MS_SYNC or MS_ASYNC shall be specified
+    // Only MS_SYNC, MS_ASYNC and MS_INVALIDATE are valid bits to be set in flag
+    if (((flags & MS_ASYNC) == MS_ASYNC) != ((flags & MS_SYNC) == MS_SYNC)
+        || (flags & (MS_SYNC | MS_ASYNC | MS_INVALIDATE)) != flags)
+        return EINVAL;
```
Comment (Dexesttp) [inline Kernel/Syscalls/mmap.cpp:599]: These comments are redundant with the code you're writing, and can be removed by writing the code in a cleaner way.

```suggestion
    if ((flags & (MS_SYNC | MS_ASYNC | MS_INVALIDATE)) != flags)
        return EINVAL;
    bool is_async = (flags & MS_ASYNC) == MS_ASYNC;
    bool is_sync = (flags & MS_SYNC) == MS_SYNC;
    if (is_async && is_sync)
        return EINVAL;
```

File: Kernel/Syscalls/mmap.cpp [hunk 2]
```diff
@@ -590,19 +590,36 @@ ErrorOr<FlatPtr> Process::sys$msyscall(Userspace<void*> address)
...
+    // Address shall be page aligned
```
Comment (Dexesttp) [inline Kernel/Syscalls/mmap.cpp:600]:
```suggestion
```

File: Kernel/Syscalls/mmap.cpp [hunk 3]
```diff
@@ -590,19 +590,36 @@ ErrorOr<FlatPtr> Process::sys$msyscall(Userspace<void*> address)
...
+    if (address.ptr() % PAGE_SIZE != 0)
+        return EINVAL;
+    // Note: This is not specified
+    size = Memory::page_round_up(size);
+
     // FIXME: We probably want to sync all mappings in the address+size range.
-    auto* region = address_space().find_region_from_range(Memory::VirtualRange { address.vaddr(), size });
+    auto* region = address_space().find_region_containing(Memory::VirtualRange { address.vaddr(), size });
+    // All regions from address upto address+size shall be mapped
```
Comment (Dexesttp) [inline Kernel/Syscalls/mmap.cpp:608]:
```suggestion
```

**Review by Dexesttp — COMMENTED** [review]
The code seems good, but the comments are redundant. See [The "Comments" section of CodingStyle.md](https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#comments) for the project's requirements about comments, in particular:
> Explain why the code does something. The code itself should already say what is happening.

File: Kernel/Syscalls/mmap.cpp (hunk 1 above)
Comment (Hendiadyoin1) [reply Kernel/Syscalls/mmap.cpp:599]: I'll go with that for now,
although I still think that having these helper variables just for a check is a bit overkill

File: Kernel/Syscalls/mmap.cpp (hunk 3 above)
Comment (Hendiadyoin1) [reply Kernel/Syscalls/mmap.cpp:608]: Not finding \*a\* region can have multiple causes, especially, once we support syncing multiple regions at once,
so the code cannot fully represent the intent as of now

**Pull request merged**